### DISCUSSION TITLE
Input Types in codegen: optional properties should be `Optional<>`

### DISCUSSION THREAD

**Discussion by tippfelher** [Netflix/dgs-framework#2054] [Q&A]
The graphqls description

```graphql
input SomeInput {
  title: String
}
```

incorrectly generates `SomeInput.java`

```java
public class SomeInput {
  String title
}
```

When `@InputArgument SomeInput input` is used in a resolver (data fetcher) there is no way to distinguish between
* the argument `input.title` was provided with value `null`
* the argument `input.title` is absent (wasn't provided by the user)

I therefore expected `SomeInput.java` to look like

```java
public class SomeInput {
  Optional<String> title
}
```

to support the use case above (distinguish between null and absence for partial updates for example)

As a workaround we can use the `DataFetchingEnvironment` to check for certain fields but this pollutes the code base and doesn't feel good.

Is there an option we can set for code generation to replace `T` with `Optional<T>` or am I misunderstanding something here?

**Comment by tippfelher** [comment]
I found a related [discussion](https://github.com/Netflix/dgs-framework/discussions/1088) and [PR](https://github.com/Netflix/dgs-framework/pull/1094), but have no idea how to use it.  😟 (I assume this is because it has nothing to do with codegen)

**Comment by tippfelher** [comment]
I finally figured it out: when **manually** implementing the Input classes the Absence of a Property can easily be determined by utilizing `Optional<T>` and setters (that is the point of the PR).

I believe this functionality should be available as a code generation option, as implementing these classes feels overly repetitive and unnecessary. Perhaps once I gain more experience with DGS and Kotlin, I’ll consider contributing a feature for it. For now, though, my issue is resolved.

Also I don't know if this feature is properly [documented](https://netflix.github.io/dgs/) or maybe I just missed it?

**Comment by paulbakker** [comment]
This was added in the latest Codegen release: https://netflix.github.io/dgs/generating-code-from-schema/#sparse-updates
